### ISSUE TITLE
compilation failed on devel branch 

### ISSUE THREAD

**Issue by vishalwc** [sysrepo/sysrepo#645]
Hi just did a git pull 
and compilation failed 

Linking CXX executable cpp_set_item_example
[ 90%] Built target cpp_set_item_example
Scanning dependencies of target Sysrepo_Lua
[ 91%] Building CXX object swig/lua/CMakeFiles/Sysrepo_Lua.dir/__/cpp/src/Sysrepo.cpp.o
[ 92%] Building CXX object swig/lua/CMakeFiles/Sysrepo_Lua.dir/__/cpp/src/Connection.cpp.o
[ 93%] Building CXX object swig/lua/CMakeFiles/Sysrepo_Lua.dir/__/cpp/src/Session.cpp.o
[ 94%] Building CXX object swig/lua/CMakeFiles/Sysrepo_Lua.dir/__/cpp/src/Struct.cpp.o
[ 95%] Building CXX object swig/lua/CMakeFiles/Sysrepo_Lua.dir/__/cpp/src/Xpath.cpp.o
[ 96%] Building CXX object swig/lua/CMakeFiles/Sysrepo_Lua.dir/__/cpp/src/Tree.cpp.o
[ 97%] Building CXX object swig/lua/CMakeFiles/Sysrepo_Lua.dir/__/cpp/src/Internal.cpp.o
Linking CXX shared library libSysrepo_Lua.so
[ 97%] Built target Sysrepo_Lua
[ 98%] Swig source
/home/vishalc/Netconf_v2/sysrepo/swig/cpp/src/Sysrepo.h:107: Error: Syntax error in input(3).
make[2]: *** [swig/lua/libsysrepoLuaLUA_wrap.cxx] Error 1
make[1]: *** [swig/lua/CMakeFiles/libsysrepoLua.dir/all] Error 2
make: *** [all] Error 2

I just removed the override key word from the reported error line and it worked

**Comment by rastislavszabo** [comment]
@mislavn could you please take a look?

**Comment by mislavn** [comment]
OK.

**Comment by mislavn** [comment]
@vishalwc I can't reproduce your bug, can you tell me how you build sysrepo (cmake command) and what Lua version have you installed locally.

**Comment by jktjkt** [comment]
This occasionally happens to me, too. I suspect that there's a breakage somewhere in CMake rules for creating the Lua bindings; apparently some files are not being regenerated upon an update.

TL;DR: remove your build dir, that helps.

Here's what the error looked like for me yesterday:

```
[34/36] Building CXX object swig/lua/CMakeFiles/libsysrepoLua.dir/libsysrepoLuaLUA_wrap.cxx.o
FAILED: /usr/bin/clang++   -DSWIG -DSWIGLUA -DlibsysrepoLua_EXPORTS -I. -I../inc -I/opt/nc/include -I../ -I../swig/cpp/src -I../swig/lua/src -I../swig/lua -fsanitize=address -Wall -Wpedantic -std=c++11 -fno-strict-aliasing -g -O0 -fPIC -MD -MT swig/lua/CMakeFiles/libsysrepoLua.dir/libsysrepoLuaLUA_wrap.cxx.o -MF swig/lua/CMakeFiles/libsysrepoLua.dir/libsysrepoLuaLUA_wrap.cxx.o.d -o swig/lua/CMakeFiles/libsysrepoLua.dir/libsysrepoLuaLUA_wrap.cxx.o -c /home/jkt/work/prog/sysrepo/build/swig/lua/libsysrepoLuaLUA_wrap.cxx
/home/jkt/work/prog/sysrepo/build/swig/lua/libsysrepoLuaLUA_wrap.cxx:11:9: warning: 'SWIGLUA' macro redefined [-Wmacro-redefined]
#define SWIGLUA
        ^
<command line>:2:9: note: previous definition is here
#define SWIGLUA 1
        ^
/home/jkt/work/prog/sysrepo/build/swig/lua/libsysrepoLuaLUA_wrap.cxx:17341:32: error: no member named 'p_error_cnt' in 'Errors'; did you mean 'error_cnt'?
    result = (size_t *)(arg1)->p_error_cnt();
                               ^~~~~~~~~~~
                               error_cnt
../swig/cpp/src/Struct.h:193:12: note: 'error_cnt' declared here
    size_t error_cnt() {return _cnt;};
           ^
/home/jkt/work/prog/sysrepo/build/swig/lua/libsysrepoLuaLUA_wrap.cxx:17377:42: error: no member named 'p_error' in 'Errors'
    result = (sr_error_info_t **)(arg1)->p_error();
                                 ~~~~~~  ^
1 warning and 2 errors generated.
ninja: build stopped: subcommand failed.
```

It seems that ``libsysrepoLuaLUA_wrap.cxx`` wasn't regenerated as it should have been.

**Comment by vishalwc** [comment]
yes after removing build , it works fine

**Comment by mislavn** [comment]
> This occasionally happens to me, too. I suspect that there's a breakage somewhere in CMake rules for creating the Lua bindings; apparently some files are not being regenerated upon an update.

@jktjkt yes I haven't yet added custom clean command to cmake I will make a pull request for this soon.

I opened a github issue for this at #648.

**Comment by rastislavszabo** [comment]
Closing this, it will be solved by #648
